src/icrar/leap-accelerate/config.h.in

```diff
@@ -1,11 +1,11 @@
-#define PROJECT_SOURCE_DIR "@PROJECT_SOURCE_DIR@/"
-#define PROJECT_BINARY_DIR "@CMAKE_BINARY_DIR@/bin/"
+#define PROJECT_SOURCE_DIR "@PROJECT_SOURCE_DIR@/" // NOLINT(cppcoreguidelines-macro-usage)
+#define PROJECT_BINARY_DIR "@CMAKE_BINARY_DIR@/bin/" // NOLINT(cppcoreguidelines-macro-usage)
 
 /// The major version of this package
-#define LEAP_ACCELERATE_VERSION_MAJOR @LEAP-Accelerate_VERSION_MAJOR@
+#define LEAP_ACCELERATE_VERSION_MAJOR @LEAP-Accelerate_VERSION_MAJOR@ // NOLINT(cppcoreguidelines-macro-usage)
 
 /// The minor version of this package
-#define LEAP_ACCELERATE_VERSION_MINOR @LEAP-Accelerate_VERSION_MINOR@
+#define LEAP_ACCELERATE_VERSION_MINOR @LEAP-Accelerate_VERSION_MINOR@ // NOLINT(cppcoreguidelines-macro-usage)
 
 /// The patch version of this package
-#define LEAP_ACCELERATE_VERSION_PATCH @LEAP-Accelerate_VERSION_PATCH@
\ No newline at end of file
+#define LEAP_ACCELERATE_VERSION_PATCH @LEAP-Accelerate_VERSION_PATCH@ // NOLINT(cppcoreguidelines-macro-usage)
```

Comment:
Isn't there a way to apply a `NOLINT` rule to a file as a whole? That way it becomes a bit less polluted with these annotations

Reply:
There are few ways to ignore some checks:
* a whole source can not be included in the call to run-clang-tidy
* headers can be ignored with headerRegexFilter in .clang-tidy
* lines ranges in files can be ignored using line filter in .clang-tidy

Reply:
I suspect using const std::string and const int should be fine here

Reply:
`version.cpp` does exactly that; having this information as macros gives you the ability to use them at compile-time on projects that might use yours, so I'd say it's better to leave these here.